app/mod_api/controllers.py

```diff
@@ -215,11 +215,22 @@ def get_resource(publisher, package, resource):
         if path.endswith('csv'):
             resource_key = metadata.build_s3_key(resource + '.csv')
             data = metadata.get_s3_object(resource_key)
-            return Response(data, mimetype='text/csv'), 200
+            def generate():
+                for row in data.splitlines():
```

Comment:
@pwalsh Noted. 
Also this made me think that `data` is already loaded in memory from s3. So there's work to be done even before that.